pkg/registry/core/service/storage/rest.go

```diff
@@ -26,41 +26,33 @@ import (
 
 	"github.com/golang/glog"
 	"k8s.io/apimachinery/pkg/api/errors"
-	metainternalversion "k8s.io/apimachinery/pkg/apis/meta/internalversion"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	utilnet "k8s.io/apimachinery/pkg/util/net"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/validation/field"
-	"k8s.io/apimachinery/pkg/watch"
 	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
 	"k8s.io/apiserver/pkg/registry/rest"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
+
 	apiservice "k8s.io/kubernetes/pkg/api/service"
 	api "k8s.io/kubernetes/pkg/apis/core"
 	"k8s.io/kubernetes/pkg/apis/core/helper"
 	"k8s.io/kubernetes/pkg/apis/core/validation"
 	"k8s.io/kubernetes/pkg/features"
-	"k8s.io/kubernetes/pkg/registry/core/endpoint"
-	podstore "k8s.io/kubernetes/pkg/registry/core/pod/storage"
+	registry "k8s.io/kubernetes/pkg/registry/core/service"
 	"k8s.io/kubernetes/pkg/registry/core/service/ipallocator"
 	"k8s.io/kubernetes/pkg/registry/core/service/portallocator"
 )
 
-// ServiceRest includes storage for services and all sub resources
-type ServiceRest struct {
-	Service *REST
-	Proxy   *ProxyREST
-}
-
 // REST adapts a service registry into apiserver's RESTStorage model.
 type REST struct {
-	registry         Registry
-	endpoints        endpoint.Registry
+	ServiceStorage
```

Comment:
don't inline this... that prevents the compiler from helping us notice if one of the methods we override changes signatures and our overridden version gets bypassed